kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/config/S3SinkConfigDef.scala

```diff
@@ -66,37 +61,6 @@ object S3SinkConfigDef
 
 }
 
-class S3SinkConfigDef() extends ConfigDef with LazyLogging {
-
-  private val processorChain: List[ConfigDefProcessor] =
-    List(new LowerCaseKeyConfigDefProcessor(CONNECTOR_PREFIX), new DeprecationConfigDefProcessor)
-
-  override def parse(jProps: util.Map[_, _]): util.Map[String, AnyRef] = {
-    val scalaProps: Map[Any, Any] = jProps.asScala.toMap
-    processProperties(scalaProps) match {
-      case Left(exception) => throw exception
-      case Right(value)    => super.parse(value.asJava)
-    }
-  }
-
-  private def processProperties(scalaProps: Map[Any, Any]): Either[Throwable, Map[Any, Any]] = {
-    val stringProps    = scalaProps.collect { case (k: String, v: AnyRef) => (k.toLowerCase, v) }
-    val nonStringProps = scalaProps -- stringProps.keySet
-    processStringKeyedProperties(stringProps) match {
-      case Left(exception)         => exception.asLeft[Map[Any, Any]]
-      case Right(stringKeyedProps) => (nonStringProps ++ stringKeyedProps).asRight
-    }
-  }
-
-  def processStringKeyedProperties(stringProps: Map[String, Any]): Either[Throwable, Map[String, Any]] = {
-    var remappedProps: Map[String, Any] = stringProps
-    for (proc <- processorChain) {
-      proc.process(remappedProps) match {
-        case Left(exception)   => return exception.asLeft[Map[String, AnyRef]]
-        case Right(properties) => remappedProps = properties
-      }
-    }
-    remappedProps.asRight
-  }
-
-}
+class S3SinkConfigDef()
+    extends CloudConfigDef(CONNECTOR_PREFIX, new DeprecationConfigDefProcessor())
+    with LazyLogging {}
```

Comment:
why not use the LazyLogging in the CloudConfigDef?

Reply:
Would that not result in the log line having `CloudConfigDef` stamped on it instead of the specific type `S3SinkConfigDef`?

Reply:
I think it will still resolve to S3SinkConfigDef

```scala
 trait LazyLogging {

  @transient
  protected lazy val logger: Logger =
    Logger(LoggerFactory.getLogger(getClass.getName))
}
```

getClass will be S3SinkConfigDef.